source/reference/configure-shell-settings-api.txt

```diff
@@ -0,0 +1,284 @@
+.. _configure-settings-api:
+
+==============================================
+Configure :binary:`~bin.mongosh` Using the API
+==============================================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+
+The ``config`` API provides methods to examine and update the
+:binary:`~bin.mongosh` configuration. Updates made using the ``config``
+API persist between sessions.
+
+.. note::
+
+   The ``config`` API can be used within the :binary:`~bin.mongosh`
+   command line interface. It has no effect in the
+   :compass:`embedded Compass shell </embedded-shell>`.
+
+Syntax
+------
+
+Print the current :binary:`~bin.mongosh` configuration:
+
+.. _mongosh-config:
+
+.. code-block:: javascript
+
+   config 
+
+Return the current value for ``<property>``:
+
+.. code-block:: javascript
+
+   config.get( "<property>" )
+
+.. _mongosh-config-set:
+
+Change the current setting of ``<property>`` to ``<value>``:
+
+.. code-block:: javascript
+
+   config.set( "<property>", <value> )
+
+Reset a ``<property>`` to the default value:
+
+.. code-block:: javascript
+
+   config.reset( "<property>" )
+
+Supported ``property`` parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/list-table-shell-properties.rst
+
+Behavior
+--------
+
+Remove or Redact Sensitive Information From History
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:binary:`~bin.mongosh` makes "best-effort" attempts to match patterns
+that normally correspond to certain kinds of sensitive information.
+
+There are
+`patterns <https://github.com/mongodb-js/redact/blob/master/lib/regexes.js>`__
+that match:
+
+- Certificates and keys
+- Generic user directories
+- Email addresses
+- IP addresses
+- URLs
+- MongoDB connection strings
+- Compass schema URL fragments
+- Electron application resources
```

Comment:
I wouldn’t expect users to care about this one in any way whatsoever (I also don’t know why exactly we’re redacting it)